volatility3/framework/layers/qemu.py

```diff
@@ -79,9 +79,10 @@ def _get_ram_segments(self, index: int, page_size: int) -> Tuple[List[Tuple[int,
             addr = self.context.object(self._qemu_table_name + constants.BANG + 'unsigned long long',
                                        offset = index,
                                        layer_name = self._base_layer)
+            # Flags are stored in the n least significant bits, where n equals the bit-length of pagesize
             flags = addr & (page_size - 1)
-            page_size_bits = int(math.log(page_size, 2))
-            addr = (addr >> page_size_bits) << page_size_bits
+            # addr equals the highest multiple of pagesize <= offset
+            addr = addr ^ (addr & (page_size - 1))
```

Comment:
This makes an implicit assumption that the `page_size` will always be a power of 2.  I don't really have a problem with that, it's almost certainly always going to be true, but it might be nice just to include a comment that we're making that assumption.  Not a showstopper, just a nice to have...  5:)

Reply:
Didn't realize that :) I will add a comment!

Reply:
Perfect, thank you!  5:D